lib/http/request/writer.rb

```diff
@@ -65,7 +65,7 @@ def send_request
         # possible in order to play nicely with Nagle's algorithm. Making
         # two writes in a row triggers a pathological case where Nagle is
         # expecting a third write that never happens.
-        data = join_headers
+        data = join_headers.b
```

Comment:
The disadvantage of `#b` is that it allocates. Since `join_headers` allocates the string anyway, it'd be more efficient to do `#force_encoding("BINARY")` here

Reply:
Basically join_headers builds a string that can be forced to be binary upfront.

Reply:
It can be actually even optimized a bit by pre-allocating string of needed length - it's pretty trivial to calculate that.

Reply:
So I should move it to using `#force_encoding("BINARY")`?

Reply:
Yes, that would be more efficient as it'd avoid duplicating the header buffer